### PR TITLE
Fix automatic conversion of constants to float32 and add support for index constant representation

### PR DESCRIPTION
**Pull request by lapid92**
Fix automatic conversion of constants to float32 and add support for index constant representation

## Checklist before requesting a review:
- [ ] I set the appropriate labels on the pull request.
- [ ] I have added/updated the release note draft (if necessary).
- [ ] I have updated the documentation to reflect my changes (if necessary).
- [ ] All function and files are well documented. 
- [ ] All function and classes have type hints.
- [ ] There is a licenses in all file.
- [ ] The function and variable names are informative. 
- [ ] I have checked for code duplications.
- [ ] I have added new unittest (if necessary).

### CODE REVIEW

File: model_compression_toolkit/core/pytorch/utils.py [hunk 1]
```diff
@@ -38,11 +38,13 @@ def set_model(model: torch.nn.Module, train_mode: bool = False):
     model.to(device)
 
 
-def to_torch_tensor(tensor):
+def to_torch_tensor(tensor,
+                    torch_dtype=np.float32):
```
Comment (elad-c) [inline model_compression_toolkit/core/pytorch/utils.py:42]: Seems like it's `numpy_type`, no `torch_dtype`. Anyway, tou can just call it `dtype`

**Review by elad-c — APPROVED** [review]
